### PR TITLE
68 test and improve precision of straight driving and turning

### PR DESCRIPTION
**Pull request by MadSoeDK**
Driving and turning using the gyro. Added PID control for driving straight and PD for turning (the "I term" not necessary atm with my tests) -> should be really great accuracy and reliability, however still limited by precision of gyro.  Let's test together monday and I'll show a demo

### CODE REVIEW

File: robot/go.sum
Comment (MadSoeDK) [inline robot/go.sum:1]: This should not be committed right? It was committed by mistake and should be removed before merge

Comment (yech199) [reply robot/go.sum:1]: @MikaelFangel

Comment (MikaelFangel) [reply robot/go.sum:1]: The go.sum should be committed (:

File: remote/src/main/java/RobotController.java [hunk 1]
```diff
@@ -30,40 +33,42 @@ public void stopController() throws InterruptedException {
...
-     * Makes the robot drive straight either forward or backwards for the distance given as argument
+     * Makes the robot drive straight either forward or backwards by using the gyro
      * @param distance Positive values in cm for forward and negative for backwards
      * @throws RuntimeException if the robot was not reached
      */
-    public void driveStraight(double distance) throws RuntimeException {
-       MultipleMotors motorsRequest = createMultipleMotorRequest(Type.l, new MotorPair(OutPort.A, DEFAULT_SPEED),
-               new MotorPair(OutPort.D, DEFAULT_SPEED));
+    public void driveWGyro(double distance) throws RuntimeException {
+        int speed = 200;
```
Comment (yech199) [inline remote/src/main/java/RobotController.java:41]: comment/document why this speed was chosen

File: remote/src/main/java/RobotController.java [hunk 2]
```diff
@@ -30,40 +33,42 @@ public void stopController() throws InterruptedException {
...
+        MultipleMotors motorsRequest = createMultipleMotorRequest(Type.l, new MotorPair(OutPort.A, speed),
+                new MotorPair(OutPort.D, speed));
 
         DriveRequest driveRequest = DriveRequest.newBuilder()
                 .setMotors(motorsRequest)
-                .setDistance((float) distance)
-                .setSpeed(DEFAULT_SPEED)
+                .setDistance((float) distance) // Note: Currently not used on the robot
+                .setSpeed(speed) // This speed worked well, other speeds could be researched
```
Comment (yech199) [inline remote/src/main/java/RobotController.java:48]: Move comment to where the speed is actually set

File: remote/src/main/java/RobotController.java [hunk 3]
```diff
@@ -30,40 +33,42 @@ public void stopController() throws InterruptedException {
...
     /**
      * Rotates the robot with itself as its center
      * @param degrees postive values for counter-clockwise and negative for clockwise
      * @throws RuntimeException if the robot was not reached
      */
-    public void rotate(double degrees) throws RuntimeException {
-        MultipleMotors motorsRequest = createMultipleMotorRequest(Type.l, new MotorPair(OutPort.A, DEFAULT_SPEED),
-                new MotorPair(OutPort.D, DEFAULT_SPEED));
+    public void rotateWGyro(double degrees) throws RuntimeException {
+        int speed = 40;
```
Comment (yech199) [inline remote/src/main/java/RobotController.java:60]: comment/document why this speed was chosen

File: remote/src/main/java/RobotController.java [hunk 4]
```diff
@@ -1,9 +1,12 @@
 import io.grpc.Grpc;
 import io.grpc.InsecureChannelCredentials;
 import io.grpc.ManagedChannel;
+import io.grpc.Status;
+import io.grpc.stub.StreamObserver;
```
Comment (yech199) [inline remote/src/main/java/RobotController.java:5]: Delete, not used

File: remote/src/main/java/RobotController.java [hunk 5]
```diff
@@ -1,9 +1,12 @@
...
 import proto.*;
 
 import java.util.ArrayList;
+import java.util.concurrent.CountDownLatch;
```
Comment (yech199) [inline remote/src/main/java/RobotController.java:9]: Delete, not used

**Comment by MikaelFangel** [comment]
Close until the PR is ready.

**Review by yech199 — COMMENTED** [review]
comments about PID a bit lacking

**Pull request closed without merging**